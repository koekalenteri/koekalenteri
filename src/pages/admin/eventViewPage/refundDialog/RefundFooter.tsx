import type { BoxProps } from '@mui/material'

import { useCallback } from 'react'
import { useTranslation } from 'react-i18next'
import { styled } from '@mui/material'
import Box from '@mui/material/Box'
import InputAdornment from '@mui/material/InputAdornment'
import Table from '@mui/material/Table'
import TableCell from '@mui/material/TableCell'
import TableRow from '@mui/material/TableRow'
import { GridFooterContainer } from '@mui/x-data-grid'

import { formatMoney, formatMoneyWithoutCurrency } from '../../../../lib/money'
import { NumberInput } from '../../../components/NumberInput'

const FooterCell = (props: BoxProps) => <Box {...props} sx={{ mx: 0 }} />

interface RefundFooterProps {
  canHaveHandlingCosts: boolean
  handlingCost: number
  onHandlingCostChange: (value?: number) => void
  total: number
  selectedTotal: number
}

declare module '@mui/x-data-grid' {
  interface FooterPropsOverrides extends RefundFooterProps {}
}

const ValueCell = styled(TableCell)(() => ({
  'padding-right': 10,
}))

export const RefundFooter = ({
  canHaveHandlingCosts,
  total,
  selectedTotal,
  handlingCost,
  onHandlingCostChange,
}: RefundFooterProps) => {
  const { t } = useTranslation()
  const parseAmount = useCallback((value: string) => {
    return Math.round(parseFloat(value.replace(',', '.')) * 100)
  }, [])
  const formatAmount = useCallback((amount: number | undefined) => formatMoneyWithoutCurrency((amount ?? 0) / 100), [])

  return (
    <GridFooterContainer>
      <FooterCell></FooterCell>
      <FooterCell>
        <Table size="small">
          <TableRow>
            <TableCell align="right">Tapahtumat yhteensä:</TableCell>
            <ValueCell align="right">{formatMoney(total / 100)}</ValueCell>
          </TableRow>
          <TableRow>
            <TableCell align="right">Käsittelykulu:</TableCell>
            <ValueCell align="right">
              <NumberInput
                disabled={!canHaveHandlingCosts}
                formatValue={formatAmount}
                parseInput={parseAmount}
                pattern={`[0-9]{1,3},[0-9]{2}`}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end" disableTypography sx={{ m: 0, color: 'primary.main' }}>
                      €
                    </InputAdornment>
                  ),
                  inputProps: {
                    style: { fontSize: 14 },
                  },
                }}
                onChange={onHandlingCostChange}
                value={handlingCost}
                variant="standard"
                error={handlingCost < 0}
                sx={{ width: 60, fontSize: 14, '.MuiInputBase-root': { fontSize: 14 } }}
              />
            </ValueCell>
          </TableRow>
          <TableRow>
            <TableCell align="right">Palautetaan:</TableCell>
            <ValueCell align="right" sx={{ color: handlingCost > selectedTotal ? 'error.main' : undefined }}>
              <strong>{formatMoney((selectedTotal - handlingCost) / 100)}</strong>
            </ValueCell>
          </TableRow>
        </Table>
      </FooterCell>
    </GridFooterContainer>
  )
}
