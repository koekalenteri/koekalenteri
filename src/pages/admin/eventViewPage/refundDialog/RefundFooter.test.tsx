import { DataGrid } from '@mui/x-data-grid'
import { render } from '@testing-library/react'

import { RefundFooter } from './RefundFooter'

describe('RefundFooter', () => {
  it('renders', async () => {
    const { container } = render(
      <DataGrid
        columns={[]}
        slots={{ footer: RefundFooter }}
        slotProps={{
          footer: {
            canHaveHandlingCosts: true,
            handlingCost: 500,
            onHandlingCostChange: jest.fn(),
            total: 1000,
            selectedTotal: 1000,
          },
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
