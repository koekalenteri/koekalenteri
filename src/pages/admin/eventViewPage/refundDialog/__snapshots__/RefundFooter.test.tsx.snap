// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefundFooter renders 1`] = `
<div>
  <div
    class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-root--noToolbar MuiDataGrid-withBorderColor css-4k60b2-MuiDataGrid-root"
    style="--DataGrid-width: 0px; --DataGrid-hasScrollX: 0; --DataGrid-hasScrollY: 0; --DataGrid-scrollbarSize: 0px; --DataGrid-rowWidth: 0px; --DataGrid-columnsTotalWidth: 0px; --DataGrid-leftPinnedWidth: 0px; --DataGrid-rightPinnedWidth: 0px; --DataGrid-headerHeight: 56px; --DataGrid-headersTotalHeight: 56px; --DataGrid-topContainerHeight: 56px; --DataGrid-bottomContainerHeight: 0px; --height: 52px;"
  >
    <div
      aria-colcount="0"
      aria-multiselectable="false"
      aria-rowcount="1"
      class="MuiDataGrid-main css-3eek4p-MuiDataGrid-main"
      role="grid"
      tabindex="-1"
    >
      <div
        class="css-qtdhae"
        data-id="gridPanelAnchor"
        role="presentation"
      />
      <div
        class="MuiDataGrid-virtualScroller css-1793420-MuiDataGrid-virtualScroller"
        role="presentation"
        style="overflow-x: hidden;"
      >
        <div
          class="MuiDataGrid-topContainer MuiDataGrid-container--top css-2sp4oa"
          role="presentation"
        >
          <div
            class="MuiDataGrid-columnHeaders css-1pmk00y-MuiDataGrid-columnHeaders"
            role="presentation"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-row--borderBottom css-yrdy0g-MuiDataGrid-columnHeaderRow"
              role="row"
            >
              <div
                role="presentation"
                style="width: 0px;"
              />
              <div
                class="MuiDataGrid-filler"
                role="presentation"
              />
              <div
                class="MuiDataGrid-scrollbarFiller MuiDataGrid-scrollbarFiller--header"
                role="presentation"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-overlayWrapper css-74bi4q-MuiDataGrid-overlayWrapper"
        >
          <div
            class="MuiDataGrid-overlayWrapperInner css-1akuw9y-MuiDataGrid-overlayWrapperInner"
            style="height: -56px; width: 0px;"
          >
            <div
              class="MuiDataGrid-overlay css-1w53k9d-MuiDataGrid-overlay"
            >
              No rows
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-virtualScrollerContent css-1kwdphh-MuiDataGrid-virtualScrollerContent"
          role="presentation"
          style="width: auto; height: 0px;"
        >
          <div
            class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
            role="rowgroup"
            style="transform: translate3d(0, 0px, 0);"
          />
        </div>
        <div
          class="MuiDataGrid-bottomContainer MuiDataGrid-container--bottom css-14egwce"
          role="presentation"
        />
      </div>
    </div>
    <div
      class="MuiDataGrid-footerContainer MuiDataGrid-withBorderColor css-wop1k0-MuiDataGrid-footerContainer"
    >
      <div
        class="MuiBox-root css-4y84eq"
      />
      <div
        class="MuiBox-root css-4y84eq"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-2ign9w-MuiTableCell-root"
              >
                Tapahtumat yhteensä:
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-4m1h4o-MuiTableCell-root"
              >
                10,00 €
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-2ign9w-MuiTableCell-root"
              >
                Käsittelykulu:
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-4m1h4o-MuiTableCell-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1uj9o6-MuiFormControl-root-MuiTextField-root"
                  inputmode="numeric"
                >
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-r5in8e-MuiInputBase-root-MuiInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-15stavw-MuiInputBase-input-MuiInput-input"
                      id=":r0:"
                      pattern="[0-9]{1,3},[0-9]{2}"
                      style="text-align: right; padding: 4px; font-size: 14px;"
                      type="text"
                      value="5,00"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeSmall css-w4mehe-MuiInputAdornment-root"
                    >
                      €
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-2ign9w-MuiTableCell-root"
              >
                Palautetaan:
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-4m1h4o-MuiTableCell-root"
              >
                <strong>
                  5,00 €
                </strong>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
